### PR TITLE
UI refresh & fixes

### PR DESCRIPTION
**Pull request by loganzartman**
Various improvements to UI:

* Overview (robots table):
  * slightly less dense with less visual noise
  * online robots above offline robots
  * robot name is now a button to open details page (no longer need to scroll table on mobile)
  * remove "watching for robots"

Before | After
--- | ---
overview_before | overview_after

* Detail view:
  * faster tab switching
    * also preserves tab state
  * less screen flashing (authorization, waiting for data) when opening detail view
  * more readable auth failure notice
  * each tab now has its own URL

### CODE REVIEW

**Comment by loganzartman** [comment]
Wouldn't it be cool to encode lat/long in status messages and show them all on a openstreetmap/mapbox map below the robots list?

File: src/App.tsx [hunk 1]
```diff
@@ -50,11 +55,10 @@ function App() {
             <IdTokenContext.Provider value={idTokenContext}>
               <CssBaseline />
               <Switch>
-                <Route exact path="/">
-                  <Overview />
-                </Route>
-                <Route exact path="/robot/:id">
-                  <Detail />
+                <Route exact path="/robot/:id/:tab?" component={Detail} />
```
Comment (kvsikand) [inline src/App.tsx:58]: Nice, tabs added to the router is a good call

**Review by kvsikand — APPROVED** [review]
Sweet, this generally looks good.
To make sure i'm following, we now always create all the tab react components, but pass them an `enabled` flag which controls whether or not they actually do things like listen for messages etc.
And then whether or not they are shown is decided by the `TabHider`. Correct?

**Comment by loganzartman** [comment]
Waiting to do a manual merge following #23 and #25

**Pull request merged**